src/main/java/de/dentrassi/rpm/builder/RpmMojo.java

```diff
@@ -842,10 +864,26 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         testLeadFlags();
 
         final BuilderOptions options = new BuilderOptions();
-        DigestAlgorithm fileDigestAlgorithm = evalDigestAlgorithm(this.fileDigestAlgorithm);
+        final DigestAlgorithm fileDigestAlgorithm = evalDigestAlgorithm(this.fileDigestAlgorithm);
         this.logger.info("File Digest Algorithm: %s", fileDigestAlgorithm.getAlgorithm());
         options.setFileDigestAlgorithm(fileDigestAlgorithm);
 
+        if (this.payloadCoding != null && !this.payloadCoding.isEmpty()) {
+            final Optional<PayloadCoding> optPayloadCoding = PayloadCoding.fromValue(this.payloadCoding);
+            final PayloadCoding payloadCoding  = optPayloadCoding.orElseThrow(() -> new MojoExecutionException("Payload coding ''" + this.payloadCoding + "' is invalid"));
+            this.logger.info("Payload Coding: %s", payloadCoding.getValue());
+            options.setPayloadCoding(payloadCoding);
```

Comment:
While the default is GZIP should the default here be ZSDT given e.g. https://fedoraproject.org/wiki/Changes/Switch_RPMs_to_zstd_compression

Reply:
I think I based it on seeing this in RPM macros:

```bash
#%_source_payload       w9.gzdio
#%_binary_payload       w9.gzdio
```

So packager defaulted to gzip 9, not even gzip default. It's possible to change the default for rpm-builder, which is separate.